tests/machine_build.rs

```diff
@@ -27,15 +23,17 @@ impl<Mac: SupportMachine> Syscalls<Mac> for SleepSyscall {
         machine.set_register(A0, Mac::REG::from_u8(0));
         Ok(true)
     }
+    fn instruction_cycles(&self, inst: ckb_vm::Instruction) -> u64 {
+        constant_cycles(inst)
+    }
 }
 
 #[cfg(has_asm)]
-pub fn asm_v1_imcb(path: &str) -> AsmMachine {
+pub fn asm_v1_imcb(path: &str) -> AsmMachine<SleepContext> {
     let buffer: Bytes = std::fs::read(path).unwrap().into();
     let asm_core = AsmCoreMachine::new(ISA_IMC | ISA_B, VERSION1, u64::max_value());
     let core = DefaultMachineBuilder::<Box<AsmCoreMachine>>::new(asm_core)
-        .instruction_cycle_func(Box::new(constant_cycles))
-        .syscall(Box::new(SleepSyscall {}))
+        .context(SleepContext)
```

Comment:
See all these places where syscalls and instruction cycles are set to the same tuple, IMO it's a strong indication that a unified context actually makes sense.